### PR TITLE
Fix/base url

### LINKED ISSUE #210: grab the baseUrl in frontend from .env
## Is your feature request related to a problem?

*Please describe.*
right now baseUrl is ```empty string```

## Describe the solution you'd like
grabbing the `baseUrl` in the Frontend from `.env`

### PR DESCRIPTION
**Pull request by palashgdev**
This issue fixes #210 

In this baseUrl is grabbed from the .env file

### CODE REVIEW

**Comment by palashgdev** [comment]
@rahulrana95 @nidhi-tandon @ankitnayan

**Comment by palashgdev** [comment]
Also we need to update the docker-compose file for the deployment as we grabbing the value from env @ankitnayan

File: frontend/webpack.config.js [hunk 1]
```diff
@@ -56,6 +63,14 @@ module.exports = {
...
 	plugins: [
 		new HtmlWebpackPlugin({ template: "src/index.html.ejs" }),
+		new webpack.ProvidePlugin({
+			process: "process",
+		}),
+		new webpack.DefinePlugin({
+			"process.env.REACT_APP_BASE_URL": JSON.stringify(
+				process.env.REACT_APP_BASE_URL,
+			),
+		}),
```
Comment (jyash97) [inline frontend/webpack.config.js:73]: How about just passing all env keys:

```suggestion
		new webpack.DefinePlugin({
			"process.env": JSON.stringify(dotenv.config().parsed),
		}),
```

File: frontend/webpack.config.js [hunk 2]
```diff
@@ -1,6 +1,13 @@
 // shared config (dev and prod)
 const { resolve } = require("path");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
+const dotenvExpand = require("dotenv-expand");
+const dotenv = require("dotenv");
+const webpack = require("webpack");
+
+const myEnv = dotenv.config();
+dotenvExpand(myEnv);
+
```
Comment (jyash97) [inline frontend/webpack.config.js:10]:
```suggestion
const dotenv = require("dotenv");
const webpack = require("webpack");
```

File: frontend/webpack.config.js [hunk 3]
```diff
@@ -56,6 +63,12 @@ module.exports = {
 	},
 	plugins: [
 		new HtmlWebpackPlugin({ template: "src/index.html.ejs" }),
+		new webpack.ProvidePlugin({
+			process: "process",
+		}),
+		new webpack.DefinePlugin({
+			"process.env": JSON.stringify(myEnv.parsed),
+		}),
```
Comment (jyash97) [inline frontend/webpack.config.js:71]:
```suggestion
		new webpack.DefinePlugin({
			"process.env": JSON.stringify(myEnv.parsed),
		}),
```

File: frontend/webpack.config.prod.js [hunk 4]
```diff
@@ -3,6 +3,12 @@ const { resolve } = require("path");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
 const CopyPlugin = require("copy-webpack-plugin");
 const CompressionPlugin = require("compression-webpack-plugin");
+const webpack = require("webpack");
+const dotenvExpand = require("dotenv-expand");
+const dotenv = require("dotenv");
+
+const myEnv = dotenv.config();
+dotenvExpand(myEnv);
```
Comment (jyash97) [inline frontend/webpack.config.prod.js:11]: Same changes as above

Comment (palashgdev) [reply frontend/webpack.config.prod.js:11]: @jyash97 process is not defined .. to do so we need to do so

File: frontend/webpack.config.js (hunk 2 above)
Comment (palashgdev) [reply frontend/webpack.config.js:10]: dotenv-expand package is there do I need to remove this one.. as it is not doing anything.. possibly we can pick this one in another issue ?

File: frontend/webpack.config.prod.js (hunk 4 above)
Comment (jyash97) [reply frontend/webpack.config.prod.js:11]: we are just defining a new variable using define plugin in this case `process.env` will point to env variables. I tried it locally it worked for me.

File: frontend/webpack.config.js (hunk 2 above)
Comment (jyash97) [reply frontend/webpack.config.js:10]: You can remove this package and update the code in the same PR.

Comment (palashgdev) [reply frontend/webpack.config.js]: @jyash97 done

File: frontend/webpack.config.js [hunk 5]
```diff
@@ -1,6 +1,11 @@
 // shared config (dev and prod)
 const { resolve } = require("path");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
+const dotenv = require("dotenv");
+const webpack = require("webpack");
+
+const myEnv = dotenv.config();
```
Comment (jyash97) [inline frontend/webpack.config.js:7]: Looks good 💯

  If we can update the variable name to `envVars` or something like this it would be good.

**Review by jyash97 — APPROVED** [review]

File: frontend/src/typings/environment.d.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+declare global {
+	namespace NodeJS {
+		interface ProcessEnv {
+			REACT_APP_BASE_URL: string;
```
Comment (nidhi-tandon) [inline frontend/src/typings/environment.d.ts:4]:
```suggestion
			FRONTEND_API_ENDPOINT: string;
```

File: frontend/webpack.config.prod.js [hunk 7]
```diff
@@ -46,12 +50,15 @@ module.exports = {
...
 	plugins: [
 		new CompressionPlugin({
-			exclude: /.map$/
+			exclude: /.map$/,
 		}),
 		new HtmlWebpackPlugin({ template: "src/index.html.ejs" }),
 		new CopyPlugin({
 			patterns: [{ from: resolve(__dirname, "public/"), to: "." }],
 		}),
+		new webpack.DefinePlugin({
```
Comment (nidhi-tandon) [inline frontend/webpack.config.prod.js:59]: Why is this required for production?

Comment (palashgdev) [reply frontend/webpack.config.prod.js:59]: then how we would grab the baseUrl?

**Comment by ankitnayan** [comment]
> Also we need to update the docker-compose file for the deployment as we grabbing the value from env @ankitnayan

@palashgdev I suggest building the image locally and install signoz with the new frontend image by changing `docker-compose.yaml`. As I last remember there was a problem in importing env variables using `process.env` inside containers. The current PR has the same problem, when you deploy the new image, frontend shows 
```
Uncaught TypeError: Cannot read property 'REACT_APP_BASE_URL' of undefined
```

**Comment by palashgdev** [comment]
> > Also we need to update the docker-compose file for the deployment as we grabbing the value from env @ankitnayan
> 
> @palashgdev I suggest building the image locally and install signoz with the new frontend image by changing `docker-compose.yaml`. As I last remember there was a problem in importing env variables using `process.env` inside containers. The current PR has the same problem, when you deploy the new image, frontend shows
> 
> ```
> Uncaught TypeError: Cannot read property 'REACT_APP_BASE_URL' of undefined
> ```

possibly the process in not defined

**Comment by palashgdev** [comment]
@ankitnayan can you please review this PR

**Review by ankitnayan — APPROVED** [review]
